src/services/download/sentry.rs

```diff
@@ -109,11 +99,17 @@ struct SearchQuery {
     token: String,
 }
 
-symbolic::common::derive_failure!(
-    SentryError,
-    SentryErrorKind,
-    doc = "Errors happening while fetching data from Sentry"
-);
+#[derive(Debug, Error)]
+pub enum SentryError {
+    #[error("failed parsing JSON response from Sentry")]
+    Parsing(JsonPayloadError),
```

Comment:
here you can even use `#[from]` instead of `#[source]` and you won't need the `.map_err()` calls in the code.

Reply:
No I can’t do neither, since `failure::Error` does not `impl std::error::Error` :-(

Reply:
I now use the `failure::Fail::compat` function to convert it to a `std::error::Error` in order to use `#[source]`.